flagsmith-jira-app/src/backend/flagsmith.ts

```diff
@@ -126,86 +107,84 @@ const unpaginate = async <TModel extends Model>(
   if (results.length === 0) throw new ApiError("Flagsmith user has no organisations", 404);
   return results;
 };
 
-export const fetchProjects = async ({
-  apiKey,
-  organisationId,
-}: {
-  apiKey: string;
-  organisationId: string | undefined;
-}): Promise<ProjectModel[]> => {
+export type ReadProjects = (args: { organisationId?: string }) => Promise<Project[]>;
+
+/** Read Flagsmith Projects for stored API Key and stored/given Organisation ID */
+export const readProjects: ReadProjects = async ({ organisationId: givenOrganisationId }) => {
+  const apiKey = await readApiKey();
   checkApiKey(apiKey);
-  if (!organisationId) throw new ApiError("Flagsmith organisation not configured", 400);
+  const organisationId = givenOrganisationId ?? (await readOrganisationId());
+  if (!organisationId) throw new ApiError("Flagsmith organisation not connected", 400);
   const path = route`/projects/?organisation=${organisationId}`;
-  const data = (await flagsmithApi(apiKey, path)) as ProjectModel[];
+  const data = (await flagsmithApi(apiKey, path)) as Project[];
   // do not unpaginate as this API does not do pagination
   const results = data ?? [];
   if (results.length === 0) throw new ApiError("Flagsmith organisation has no projects", 404);
   return results;
 };
 
-export const fetchEnvironments = async ({
-  apiKey,
-  projectId,
-}: {
-  apiKey: string;
-  projectId: string | undefined;
-}): Promise<EnvironmentModel[]> => {
+export type ReadEnvironments = (args: { projectId?: string }) => Promise<Environment[]>;
+
+/** Read Flagsmith Environments for stored API Key and given Project ID */
+export const readEnvironments: ReadEnvironments = async ({ projectId }) => {
+  const apiKey = await readApiKey();
   checkApiKey(apiKey);
-  if (!projectId) throw new ApiError("Flagsmith project not configured", 400);
+  if (!projectId) throw new ApiError("Flagsmith project not connected", 400);
   const path = route`/environments/?project=${projectId}`;
-  const data = (await flagsmithApi(apiKey, path)) as PaginatedModels<EnvironmentModel>;
+  const data = (await flagsmithApi(apiKey, path)) as PaginatedModels<Environment>;
   const results = await unpaginate(apiKey, data);
   if (results.length === 0) throw new ApiError("Flagsmith project has no environments", 404);
   return results;
 };
 
-export const fetchFeatures = async ({
-  apiKey,
-  projectId,
-}: {
-  apiKey: string;
-  projectId: string | undefined;
-}): Promise<FeatureModel[]> => {
+export type ReadFeatures = (args: { projectId?: string }) => Promise<Feature[]>;
+
+/** Read Flagsmith Features for stored API Key and given Project ID */
+export const readFeatures: ReadFeatures = async ({ projectId }) => {
+  const apiKey = await readApiKey();
   checkApiKey(apiKey);
-  if (!projectId) throw new ApiError("Flagsmith project not configured", 400);
+  if (!projectId) throw new ApiError("Flagsmith project not connected", 400);
   const path = route`/projects/${projectId}/features/`;
```

Comment:
I have a feeling that the fix for the identity / segment badges is as 'simple' as: 

```suggestion
  const path = route`/projects/${projectId}/features/?environment={environmentId}`;
```

Obviously, the complication is making sure that we have the integer id that's needed here available to us, and passing it into this function. I'm happy to tackle this as a separate PR, but it might be worth doing now-ish while we're already working on the JIRA integration @riceyrice .

Reply:
OK done it 😁